cmd/kpod/start.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"os"
+	"strconv"
+)
+
+var (
+	startFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "attach, a",
+			Usage: "Attach container's STDOUT and STDERR",
+		},
+		cli.StringFlag{
+			Name:  "detach-keys",
+			Usage: "Override the key sequence for detaching a container. Format is a single character [a-Z] or ctrl-<value> where <value> is one of: a-z, @, ^, [, , or _.",
+		},
+		cli.BoolFlag{
+			Name:  "interactive, i",
+			Usage: "Keep STDIN open even if not attached",
+		},
+	}
+	startDescription = `
+   kpod start
+
+   Starts one or more containers.  The container name or ID can be used.
+`
+
+	startCommand = cli.Command{
+		Name:        "start",
+		Usage:       "Start one or more containers",
+		Description: startDescription,
+		Flags:       startFlags,
+		Action:      startCmd,
+		ArgsUsage:   "CONTAINER-NAME [CONTAINER-NAME ...]",
+	}
+)
+
+func startCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) < 1 {
+		return errors.Errorf("you must provide at least one container name or id")
+	}
+
+	attach := c.Bool("attach")
+
+	if len(args) > 1 && attach {
+		return errors.Errorf("you cannot start and attach multiple containers at once")
+	}
+
+	if err := validateFlags(c, startFlags); err != nil {
+		return err
+	}
+
+	runtime, err := getRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	var lastError error
+	for _, container := range args {
+		// Create a bool channel to track that the console socket attach
+		// is successful.
+		attached := make(chan bool)
+		// Create a waitgroup so we can sync and wait for all goroutines
+		// to finish before exiting main
+		var wg sync.WaitGroup
+
+		noStdIn := true
+		ctr, err := runtime.LookupContainer(container)
+		if err != nil {
+			lastError = errors.Wrapf(err, "unable to find container %s", container)
+			fmt.Fprintln(os.Stderr, lastError)
```

Comment:
You should be checking if lastError is set, then print it and replace lastError with new last error.

Reply:
fixed